samples/wifi/shell/src/wifi_shell.c

```diff
@@ -0,0 +1,341 @@
+			str_clear_free(ssid->sae_password);
+			ssid->sae_password = dup_binstr(params->psk, params->psk_length);
+
+			if (ssid->sae_password == NULL) {
+				shell_fprintf(context.shell,
+					      SHELL_ERROR,
+					      "%s:Failed to copy sae_password\n",
+					      __func__);
+				return -1;
+			}
+		} else {
+			if (params->security == 2)
+				ssid->key_mgmt = WPA_KEY_MGMT_PSK_SHA256;
+			else
+				ssid->key_mgmt = WPA_KEY_MGMT_PSK;
+
+			str_clear_free(ssid->passphrase);
+			ssid->passphrase = dup_binstr(params->psk, params->psk_length);
+
+			if (ssid->passphrase == NULL) {
+				shell_fprintf(context.shell,
+					      SHELL_ERROR,
+					      "%s:Failed to copy passphrase\n",
+					      __func__);
+				return -1;
+			}
+		}
+
+		wpa_config_update_psk(ssid);
+
+		if (pmf)
+			ssid->ieee80211w = 1;
+
+	}
+
+	wpa_supplicant_enable_network(wpa_s_0,
+				      ssid);
+
+	wpa_supplicant_select_network(wpa_s_0,
+				      ssid);
+
+	return 0;
+}
+
+
+static int cmd_supplicant(const struct shell *shell,
+			  size_t argc,
+			  const char *argv[])
+{
+	return cli_main(argc,
+			argv);
+}
+#endif /* CONFIG_WPA_SUPP */
+
+
+SHELL_STATIC_SUBCMD_SET_CREATE(
+	wpa_cli_cmds,
+	SHELL_CMD(scan,
+		  NULL,
+		  "Scan AP",
+		  cmd_supplicant_scan),
+#ifdef CONFIG_WPA_SUPP
+	SHELL_CMD(connect,
+		  NULL,
+		  " \"<SSID>\""
+		  "\nPassphrase (optional: valid only for secured SSIDs)>"
+		  "\nKEY_MGMT (optional: 0-None, 1-WPA2, 2-WPA2-256, 3-WPA3)",
+		  cmd_supplicant_connect),
+	SHELL_CMD(add_network,
+		  NULL,
+		  "\"Add Network network id/number\"",
+		  cmd_supplicant),
+	SHELL_CMD(set_network,
+		  NULL,
+		  "\"Set Network params\"",
+		  cmd_supplicant),
+	SHELL_CMD(set,
+		  NULL,
+		  "\"Set Global params\"",
+		  cmd_supplicant),
+	SHELL_CMD(get,
+		  NULL, "\"Get Global params\"",
+		  cmd_supplicant),
+	SHELL_CMD(enable_network,
+		  NULL,
+		  "\"Enable Network\"",
+		  cmd_supplicant),
+	SHELL_CMD(remove_network,
+		  NULL,
+		  "\"Remove Network\"",
+		  cmd_supplicant),
+	SHELL_CMD(get_network,
+		  NULL,
+		  "\"Get Network\"",
+		  cmd_supplicant),
+	SHELL_CMD(select_network,
+		  NULL,
+		  "\"Select Network which will be enabled; rest of the networks will be disabled\"",
+		  cmd_supplicant),
+	SHELL_CMD(disable_network,
+		  NULL,
+		  "\"\"",
+		  cmd_supplicant),
+	SHELL_CMD(disconnect,
+		  NULL,
+		  "\"\"",
+		  cmd_supplicant),
+	SHELL_CMD(reassociate,
+		  NULL,
+		  "\"\"",
+		  cmd_supplicant),
+	SHELL_CMD(status,
+		  NULL,
+		  "\"Get client status\"",
+		  cmd_supplicant),
+	SHELL_CMD(bssid,
+		  NULL,
+		  "\"Associate with this BSSID\"",
+		  cmd_supplicant),
+	SHELL_CMD(sta_autoconnect,
+		  NULL,
+		  "\"\"",
+		  cmd_supplicant),
+	SHELL_CMD(signal_poll,
+		  NULL,
+		  "\"\"",
+		  cmd_supplicant),
+#endif /* CONFIG_WPA_SUPP */
+	SHELL_SUBCMD_SET_END);
+
+
+SHELL_CMD_REGISTER(wpa_cli,
+		   &wpa_cli_cmds,
+		   "WPA supplicant commands",
+		   NULL);
+
+
+static int supplicant_shell_init(const struct device *unused)
```

Comment:
Function name is incosistent with the module name
```suggestion
static int wifi_shell_init(const struct device *unused)
```